nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/WOPIMapWriter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thomas Roger
+ */
+
+package org.nuxeo.wopi.jaxrs;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Provider;
+
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+
+/**
+ * Writer writing a {@link WOPIMap} object.
+ *
+ * @since 10.3
+ */
+@Provider
+@Produces(MediaType.APPLICATION_JSON)
+public class WOPIMapWriter implements MessageBodyWriter<WOPIMap> {
+
+    @Context
+    protected JsonFactory factory;
+
+    @Override
+    public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return WOPIMap.class.isAssignableFrom(type);
+    }
+
+    @Override
+    public long getSize(WOPIMap wopiMap, Class<?> type, Type genericType, Annotation[] annotations,
+            MediaType mediaType) {
+        return -1;
+    }
+
+    @Override
+    public void writeTo(WOPIMap wopiMap, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType,
+            MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream)
+            throws IOException, WebApplicationException {
+        JsonGenerator jg = factory.createGenerator(entityStream, JsonEncoding.UTF8);
```

Comment:
Yes, will push it in a Sonar fixes commit.

Reply:
I'm only mentioning it because it's the only Blocker bug reported by Sonar at the moment.
https://sonarcloud.io/project/issues?id=org.nuxeo%3Anuxeo-ecm&resolved=false&severities=BLOCKER&types=BUG